src/backend/distributed/executor/multi_utility.c

```diff
@@ -572,6 +572,23 @@ ProcessAlterTableStmt(AlterTableStmt *alterTableStatement, const char *alterTabl
 static void
 ErrorIfUnsupportedIndexStmt(IndexStmt *createIndexStatement)
 {
+	Oid namespaceId;
+	char* indexRelationName = createIndexStatement->idxname;
+
+	if (indexRelationName == NULL)
+	{
+		ereport(ERROR, (errcode(ERRCODE_DUPLICATE_TABLE),
+										errmsg("you must specify a name when creating an index "
+													 "on a distributed table")));
+	}
+
+	namespaceId = get_namespace_oid(createIndexStatement->relation->schemaname, false);
+	if (get_relname_relid(indexRelationName, namespaceId))
```

Comment:
I could not relate not finding a relationId with already exists error. Could you provide a description here ?